### PR TITLE
ML Engine: Versions on Ansible

### PR DESCRIPTION
**Pull request by rambleraptor**
https://github.com/GoogleCloudPlatform/magic-modules/issues/2061

Versions on Ansible! This also has a couple Model fixes:
* Async on Model Deletes (because just deletes are async on Models)
* A better decoder that doesn't break on Operations.

# Release Note for Downstream PRs (will be copied)
```releasenote

```

### CODE REVIEW

File: products/mlengine/api.yaml [hunk 1]
```diff
@@ -73,3 +91,148 @@ objects:
...
+      - !ruby/object:Api::Type::Enum
+        name: 'framework'
+        description: The machine learning framework AI Platform uses to train this version of the model
+        values:
+          - FRAMEWORK_UNSPECIFIED
+          - TENSORFLOW
+          - SCIKIT_LEARN
+          - XGBOOST
+      - !ruby/object:Api::Type::String
+        name: 'pythonVersion'
```
Comment (SirGitsalot) [inline products/mlengine/api.yaml]: If there's a fixed set of values for this param, should it be an Enum?

File: products/mlengine/api.yaml [hunk 2]
```diff
@@ -73,3 +91,148 @@ objects:
...
+          Automatically scale the number of nodes used to serve the model in
+          response to increases and decreases in traffic. Care should be taken
+          to ramp up traffic according to the model's ability to scale or you
+          will start seeing increases in latency and 429 response codes.
+        conflicts: ['manualScaling']
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'minNodes'
+            description: The minimum number of nodes to allocate for this mode
+      - !ruby/object:Api::Type::NestedObject
```
Comment (SirGitsalot) [inline products/mlengine/api.yaml]: Is it possible to flatten the scaling-related nested objects' single member fields into top-level params?

File: products/mlengine/api.yaml (hunk 1 above)
Comment (rambleraptor) [reply products/mlengine/api.yaml]: fixed!

File: products/mlengine/api.yaml (hunk 2 above)
Comment (rambleraptor) [reply products/mlengine/api.yaml]: Ansible doesn't have this feature currently (I believe TF does).

**Comment by rambleraptor** [comment]
I also added the support to set a default version (uses a custom endpoint).

**Pull request merged**